> > @soehms Can you help me test this?
> 
> See https://github.com/soehms/sage/actions/runs/25510035076/job/74866062978

Now, #42138 is fixed:


```
docker run -it sagemath/sagemath:10.9.patch9
Unable to find image 'sagemath/sagemath:10.9.patch9' locally
10.9.patch9: Pulling from sagemath/sagemath
...
29d36b1c863f: Pull complete 
Digest: sha256:88c1e8bace1f1bec09712b46ee237db246326faae6dec8b0c95b234669357d32
Status: Downloaded newer image for sagemath/sagemath:10.9.patch9
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 10.9, Release Date: 2026-05-04                    │
│ Using Python 3.12.3. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
```

while #40867 is not:

```
docker run -it sagemath/sagemath:10.9.patch9 /bin/bash
sage@31e1b281f004:~$ sage
Traceback (most recent call last):
  File "/home/sage/sage/src/bin/sage-ipython", line 9, in <module>
    from sage.misc.banner import banner
ModuleNotFoundError: No module named 'sage'
sage@893228cdcdd8:~$ echo $PATH
/home/sage/sage/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

@cxzhong, you should change the header comment accordingly!

If you execute bash in the container created upon sagemath/sagemath:10.9.patch9 without arguments, Sage works even though `$PATH` looks the same:

```
sebastian@ThinkPadKlein:~/devel/github/database_knotinfo$ docker exec -it d51c151546ab bash
sage@d51c151546ab:~$ sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 10.9, Release Date: 2026-05-04                    │
│ Using Python 3.12.3. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
sage: quit
sage@d51c151546ab:~$ echo $PATH
/home/sage/sage/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

I think if we can quickly find a solution for issue #40867, we should include it. Otherwise, this would be another pull request.

We should also consider adding a third job to `.github/workflows/push_to_docker_hub.yml` (with `needs: sagemath`) that checks if the image is functional.

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -87,7 +87,6 @@ RUN ln -s "$SAGE_ROOT/sage" /usr/bin/sage
 RUN ln -s /usr/bin/sage /usr/bin/sagemath
 # Sage needs the fortran libraries at run-time because we do not build gfortran
 # with Sage but use the system's.
-# Sage's virtual environment uses the system Python executable at run-time.
 # We need gcc/g++ to allow compilation of cython at run-time from the notebook.
 # We also install sudo for the sage user, see below.
 RUN apt-get -qq update \
@@ -233,7 +232,7 @@ ARG SAGE_ROOT=/home/sage/sage
 COPY --chown=sage:sage --from=make-release $SAGE_ROOT/ $SAGE_ROOT/
 # set path in bashrc to start gap, gp, maxima, ... this does not
 # affect Sage (see https://github.com/sagemath/sage/pull/38049)
-RUN echo "export PATH=$SAGE_ROOT/local/bin:$PATH" >> ~/.bashrc
+RUN echo 'export PATH='"$SAGE_ROOT"'/local/bin:$PATH' >> ~/.bashrc
 
 COPY ./docker/entrypoint.sh /usr/local/bin/sage-entrypoint
 WORKDIR $HOME

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -90,7 +90,7 @@ RUN ln -s /usr/bin/sage /usr/bin/sagemath
 # We need gcc/g++ to allow compilation of cython at run-time from the notebook.
 # We also install sudo for the sage user, see below.
 RUN apt-get -qq update \
-    && apt-get -qq install -y --no-install-recommends gfortran gcc g++ sudo openssl pkg-config patch libz-dev libbz2-dev bzip2 ca-certificates perl libboost-dev python3 \
+    && apt-get -qq install -y --no-install-recommends gfortran gcc g++ sudo openssl pkg-config patch libz-dev libbz2-dev bzip2 ca-certificates perl libboost-dev python3 python3-dev \
     && apt-get -qq clean \
     && rm -r /var/lib/apt/lists/*
 # Sage refuses to build as root, so we add a "sage" user that can sudo without a password.

